lib/smart_name/manipulate.rb

```diff
@@ -0,0 +1,73 @@
+class SmartName
+  module Manipulate
+    # replace a subname
+    # keys are used for comparison
+    def replace old, new
+      old_name = old.to_name
+      new_name = new.to_name
+      return self if old_name.length > length
+      return replace_part(old_name, new_name) if old_name.simple?
+      return self unless include? old_name
+      replace_all_subsequences(old_name, new_name).to_name
+    end
```

Comment:
I think you know I was joking when I said that anyone who moves code has to refactor it, right?

But this _rocks_!